src/executables/srpd_rotation.c

```diff
@@ -0,0 +1,398 @@
+        SRPLG_LOG_ERR("srpd_rotation", "Fatal error: Archive directory could not be created");
+        goto cleanup;
+    }
+
+    for ( ; ATOMIC_LOAD_RELAXED(opts->running); sleep(1)) {
+
+        /* Remember current time */
+        time(&current_time);
+
+        /* Open directory */
+        d = opendir(notif_dir_name);
+        if (!d) {
+            continue;
+        }
+
+        /* Read whole directory */
+        while ((dir = readdir(d)) && ATOMIC_LOAD_RELAXED(opts->running)) {
+
+            /* Skip current and parent directories */
+            if (!strcmp(dir->d_name, ".") || !strcmp(dir->d_name, "..")) {
+                continue;
+            }
+
+            /* Check correct format of the file and retrieve file times */
+            if (srpd_format_check(dir->d_name, NULL, &file_time2)) {
+                continue;
+            }
+
+            /* Check whether a file is older than configured time */
+            if ((current_time >= (time_t)ATOMIC_LOAD_RELAXED(opts->rotation_time)) &&
+                    (file_time2 < (current_time - (time_t)ATOMIC_LOAD_RELAXED(opts->rotation_time)))) {
+
+                /* Build zipping args */
+                if (ATOMIC_LOAD_RELAXED(opts->compress)) {
+                    if (asprintf(&arg1, "%s%s.zip", (char *)ATOMIC_PTR_LOAD_RELAXED(opts->output_folder), dir->d_name) == -1) {
+                        goto cleanup;
+                    }
+                    if (asprintf(&arg2, "%s%s", notif_dir_name, dir->d_name) == -1) {
+                        goto cleanup;
+                    }
+
+                    /* Zip a file in output folder */
+                    if ((rc = srpd_exec("srpd_rotation", SRPD_ZIP_BINARY, 3, SRPD_ZIP_BINARY, arg1, arg2))) {
+                        SRPLG_LOG_ERR("srpd_rotation", "Zipping a file failed.");
+                    } else {
+                        ATOMIC_INC_RELAXED(opts->rotated_files_count);
+
+                        if (asprintf(&remove_str, "%s%s", notif_dir_name, dir->d_name) == -1) {
+                            goto cleanup;
+                        }
+
+                        /* Remove a file from notif folder */
+                        if (remove(remove_str)) {
+                            SRPLG_LOG_ERR("srpd_rotation", "Removing a file failed.");
+                        }
+                    }
+
+                    /* Build moving args */
+                } else {
+                    if (asprintf(&arg1, "%s%s", notif_dir_name, dir->d_name) == -1) {
+                        goto cleanup;
+                    }
+                    if (asprintf(&arg2, "%s%s", (char *)ATOMIC_PTR_LOAD_RELAXED(opts->output_folder), dir->d_name) == -1) {
+                        goto cleanup;
+                    }
+
+                    /* Move a file to the output folder */
+                    if ((rc = rename(arg1, arg2)) == -1) {
+                        SRPLG_LOG_ERR("srpd_rotation", "Moving a file failed.");
```

Comment:
Include the moved file name in the error.

Reply:
Done